sdk/core/scheduler/thread.h

```diff
@@ -512,11 +564,25 @@ namespace sched
 			/// right after waking up (but before clearing the bits if
 			/// clearOnExit is set). 0 if it's woken up due to timer expiry.
 			EventBits eventWaitBits;
-			/**
-			 * If this thread is blocked on a futex, this holds the address of
-			 * the futex.  This is set to 0 if woken via timeout.
-			 */
-			ptraddr_t futexWaitAddress;
+			/// State associated with waiting on a futex.
+			struct
+			{
+				/**
+				 * If this thread is blocked on a futex, this holds the address
+				 * of the futex.  This is set to 0 if woken via timeout.
+				 */
+				ptraddr_t futexWaitAddress;
+				/**
+				 * The thread that we're priority boosting.  This is ignored if
+				 * `futexPriorityInheriting` is false.
+				 */
+				uint16_t futexPriorityBoostingThread : 16;
```

Comment:
Ah, I realized part of my confusion above: I was sometimes reading this with subject and object switched, as the thread that's boosting us.  Maybe (⚠️:🚲🏠) `futexPriorityBoostedThread` instead?